app.rb

```diff
@@ -1,9 +1,33 @@
 require 'sinatra/base'
+require './lib/peep'
 
 class Chitter < Sinatra::Base
+  enable :sessions
+
   get '/test' do
     'Test page'
   end
 
+  get '/peeps' do
+    @peeps = Peep.all
+    erb(:index)
+  end
+
+  post '/peeps' do
+    Peep.add(username: params[:peep_user], message: params[:peep_message])
+    redirect '/peeps'
+  end
+
```

Comment:
There is a way of adding search functionality to the index action `GET "/posts"`.  It would look something like this...

```ruby
get "/posts" do
  if params[:search_keyword]
    @peeps = Peep.search(params[:search_keyword])
  else
    @peeps = Peep.all
  end

  erb :index
end
```

You could argue that this breaks SRP and you might have a point. I think, however, I'm OK with it since either way the user ends up with a list of some `Peep`s.

Alternatively, you could still simplify things a little bit by using the same `index` template for both search results and showing all peeps.

Either way, I think the request generated when someone submits the search form would be a GET request, since nothing is being added to the DB. It's just a GET request with some params. Similar to if you wanted to see one single peep, which would be found using it's ID, passed in the URL.